force_torque_sensor_broadcaster/src/force_torque_sensor_broadcaster.cpp

```diff
@@ -141,23 +141,97 @@ controller_interface::CallbackReturn ForceTorqueSensorBroadcaster::on_deactivate
     realtime_publisher_->unlockAndPublish();
   }
 
   return controller_interface::return_type::OK;
 }
 
+controller_interface::return_type ForceTorqueSensorBroadcaster::update_reference_from_subscribers(
+  const rclcpp::Time & /*time*/, const rclcpp::Duration & /*period*/)
+{
+  return controller_interface::return_type::OK;
+}
+
+std::vector<hardware_interface::StateInterface>
+ForceTorqueSensorBroadcaster::on_export_state_interfaces()
+{
+  std::vector<hardware_interface::StateInterface> exported_state_interfaces;
+
+  std::vector<std::string> force_names(
+    {params_.interface_names.force.x, params_.interface_names.force.y,
+     params_.interface_names.force.z});
+  std::vector<std::string> torque_names(
+    {params_.interface_names.torque.x, params_.interface_names.torque.y,
+     params_.interface_names.torque.z});
+  if (!params_.sensor_name.empty())
+  {
+    const auto semantic_comp_itf_names = force_torque_sensor_->get_state_interface_names();
+    std::copy(
+      semantic_comp_itf_names.begin(), semantic_comp_itf_names.begin() + 3, force_names.begin());
+    std::copy(
+      semantic_comp_itf_names.begin() + 3, semantic_comp_itf_names.end(), torque_names.begin());
+  }
+  const std::string controller_name = get_node()->get_name();
+  if (!force_names[0].empty())
+  {
+    exported_state_interfaces.emplace_back(hardware_interface::StateInterface(
+      controller_name, force_names[0], &realtime_publisher_->msg_.wrench.force.x));
+  }
+  if (!force_names[1].empty())
+  {
+    exported_state_interfaces.emplace_back(hardware_interface::StateInterface(
+      controller_name, force_names[1], &realtime_publisher_->msg_.wrench.force.y));
+  }
+  if (!force_names[2].empty())
+  {
+    exported_state_interfaces.emplace_back(hardware_interface::StateInterface(
+      controller_name, force_names[2], &realtime_publisher_->msg_.wrench.force.z));
+  }
+  if (!torque_names[0].empty())
+  {
+    exported_state_interfaces.emplace_back(hardware_interface::StateInterface(
+      controller_name, torque_names[0], &realtime_publisher_->msg_.wrench.torque.x));
+  }
+  if (!torque_names[1].empty())
+  {
+    exported_state_interfaces.emplace_back(hardware_interface::StateInterface(
+      controller_name, torque_names[1], &realtime_publisher_->msg_.wrench.torque.y));
+  }
+  if (!torque_names[2].empty())
+  {
+    exported_state_interfaces.emplace_back(hardware_interface::StateInterface(
+      controller_name, torque_names[2], &realtime_publisher_->msg_.wrench.torque.z));
+  }
+  return exported_state_interfaces;
+}
+
+void ForceTorqueSensorBroadcaster::apply_sensor_offset(
+  const Params & params, geometry_msgs::msg::WrenchStamped & msg)
+{
+  msg.wrench.force.x -= params.offset.force.x;
+  msg.wrench.force.y -= params.offset.force.y;
+  msg.wrench.force.z -= params.offset.force.z;
+  msg.wrench.torque.x -= params.offset.torque.x;
+  msg.wrench.torque.y -= params.offset.torque.y;
+  msg.wrench.torque.z -= params.offset.torque.z;
```

Comment:
I don't want to be a nag but I'd reeeeeeeally expect offsets to be *added* using a `+`

Reply:
A very very fair point!
I'll try to change it ASAP

Reply:
Done. I've made the changes. Thanks for the feedback

Reply:
thank you :D